### DISCUSSION TITLE
Finding the state for a caselaw

### DISCUSSION THREAD

**Discussion by amit-lf** [freelawproject/courtlistener#4276] [General]
Hi,  I want to find out which state a case law belongs to. I was planning to use court_id as a surrogate till I realized that court_ids like ca1 serve multiple states. If it is possible to go more granular like the city or zipcode that will be great.  Help is appreciated. Thanks

**Comment by mlissner** [comment]
This is a tough one. You might be able to get some of this from FJC's integrated DB, but that's only federal cases. We import that data from time to time, but haven't done so for a few years.

**Comment by legaltextai** [comment]
Although not updated anymore, https://huggingface.co/datasets/harvard-lil/cold-cases has the state information. 
@mlissner Mike, you used to have a page dedicated to court labels / names, like 'ind' is Indiana Supreme Court, but I can't find that anymore. 
There should be a way to train the model to deduce the state. How did Harvard added the court jurisdiction column?

**Reply by mlissner** [reply]
Some states are easy, like Indiana Supreme Couirt. Others are impossible like CA9. No idea if/how CAP did/does it.

The page you're looking for is: https://www.courtlistener.com/help/api/jurisdictions/, I think?

**Reply by amit-lf** [reply]
Thanks Mike, 'legaltextai'.  

Thanks for sharing the helpful URL.  Agree that in some cases it is easy to infer but thanks for confirming the difficulty in cases like CA9.

**Comment by legaltextai** [comment]
attached is a court_short_name,court_full_name,court_jurisdiction based on harvard_cold dataset 
you can join it into the court CL dataset 
i've already noticed some errors or incomplete information, but it is a good starting point if you need to add 'court_jurisdiction' like e.g. "Ohio, OH" and then filter out by states 
hope this helps 
[court_names_jurisdiction.csv](https://github.com/user-attachments/files/16529129/court_names_jurisdiction.csv)

**Reply by mlissner** [reply]
Hm, it's new, not fully deployed, and only partly populated, but we actually have a courthouse table that's joined to the Court table. I'd welcome a PR to include it as nested and filterable data in our Court API if anybody was interested.

**Comment by legaltextai** [comment]
i think courts like ca1, ca2 etc would fall under "USA, Federal" jurisdiction, not state jurisdiction

**Reply by BuffaloJames** [reply]
The court where the case originated could be of interest to some researchers but difficult to deduce. An example could be a federal constitutional challenge to a statute in state X. That case would be filed in federal court, then could make its way to a federal appellate court, and eventually the US Supreme Court. To know that the case originated in state X, one would need to read the facts in the opinions from the various courts.

An easier, though less complete, solution could be what others above said and map the state courts and call the other ones federal.
